compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -277,9 +289,49 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             }
         };
         imm = bx.to_immediate_scalar(imm, to_scalar);
+        self.assume_scalar_range(bx, imm, to_scalar, to_backend_ty);
```

Comment:
is that necessary? I'd assume that part is obvious to LLVM, as it's the type of the value you get. If there is a reason this isn't covered by LLVM, please document it here.

Reply:
If this is a `transmute` to `char`, for example, the LLVM type is just `i32`, so there's nothing on it to say that it's limited.  I'll add a note.